### PR TITLE
Denver "Puplic" Library - Pull Request

### PR DESCRIPTION
**Pull request by TristanB17**
I feel as though I performed much better on this test, mostly because I've been practicing those kind of 'code for both tests and implementation' exercises. I really feel as though I was close to completing iteration 4, but something was amiss with my 'include?' method (which I thought would be relatively simple to figure out).

### CODE REVIEW

File: test/author_test.rb [hunk 1]
```diff
@@ -1,6 +1,48 @@
...
+  end
+
+  def test_author_can_hold_multiple_books
+    charlotte_bronte = Author.new({
+      first_name: "Charlotte",
+      last_name: "Bronte"
+      })
+
+    charlotte_bronte.add_book("Jane Eyre", "October 16, 1847")
+    assert_instance_of Book, charlotte_bronte.books[0]
```
Comment (icorson3) [inline test/author_test.rb]: this seems a bit redundant. would be happy if this was testing once and length was continuously tested.

File: lib/library.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize
+    @books = []
+  end
+
+  def add_to_collection(book)
+    @books << book
+  end
+
+  def include?(work)
+    @books.any?{|book| book.title == work}
```
Comment (icorson3) [inline lib/library.rb]: got close here...try starting with TDD to get this up and running. a solution such as `@books.include?(work)` would have worked

File: test/author_test.rb [hunk 3]
```diff
@@ -1,6 +1,48 @@
...
+      })
+    assert_equal [], charlotte_bronte.books
+  end
+
+  def test_a_book_can_be_added_to_author
+    charlotte_bronte = Author.new({
+      first_name: "Charlotte",
+      last_name: "Bronte"
+      })
+    charlotte_bronte.add_book("Jane Eyre", "October 16, 1847")
```
Comment (icorson3) [inline test/author_test.rb]: Another way to accomplish this test is to use the return value of the method `add_book`. `add_book` will return a book object so I can verify that the book in the books array is "Jane Eyre". so an assertion that would be good might be `jane_eyre = charlotte_bronte.add_book("Jane Eyre", "October 16, 1847") assert_equal jane_eyre, charlotte_bronte.books[0]`

**Comment by icorson3** [comment]
@TristanB17 left some comments...would love to see you work on testing more!

File: test/book_test.rb [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+
+  def test_a_book_has_attributes
+    book = Book.new({
+      author_first_name: "Harper",
+      author_last_name: "Lee",
+      title: "To Kill a Mockingbird",
+      publication_date: "July 11, 1960"
+      })
+    assert_equal "Harper", book.author_first_name
+    assert_equal "To Kill a Mockingbird", book.title
```
Comment (BrianZanti) [inline test/book_test.rb]: You need to test all the attributes

File: test/library_test.rb [hunk 5]
```diff
@@ -1,6 +1,69 @@
...
+      last_name: "Bronte"
+      })
+    jane_eyre = charlotte_bronte.add_book("Jane Eyre", "October 16, 1847")
+    villette  = charlotte_bronte.add_book("Villette", "1853")
+
+    harper_lee  = Author.new({
+      first_name: "Harper",
+      last_name: "Lee"
+      })
+    mockingbird = harper_lee.add_book("To Kill a Mockingbird", "July 11, 1960")
```
Comment (BrianZanti) [inline test/library_test.rb]: This book is never used.

**Pull request closed without merging**